src/components/DataTable/TableToolbarDownload.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file Table toolbar button for downloading table data as CSV
+ * @copyright IBM Security 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import Download16 from '@carbon/icons-react/lib/download/16';
+import { ExportToCsv } from 'export-to-csv';
+import IconButton from '../IconButton';
+
+const options = {
+  fieldSeparator: ',',
+  quoteStrings: '"',
+  decimalSeparator: '.',
+  showLabels: true,
+  useTextFile: false,
+  useBom: true,
+};
+
+const TableToolbarDownload = ({ headers, rows, title, filename }) => {
+  const csvRows = rows.map(row => {
+    const newRow = {};
+    headers.forEach(header => {
+      newRow[header.key] = row[header.key];
+    });
+    return newRow;
+  });
+
+  if (title) {
+    options.showTitle = true;
+    options.title = title;
+  }
+  options.filename = filename;
+
+  options.headers = headers.map(head => head.header);
```

Comment:
- [ ] You can destructure `header` here to reduce duplicating `head` - 

```suggestion
  options.headers = headers.map(({ header }) => header);
```